test/kotlin/com/example/people_managment_system/controller/SystemControllerTest.kt

```diff
@@ -0,0 +1,308 @@
+package com.example.people_managment_system.controller
+
+import com.example.people_managment_system.dtos.EmployeeDto
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.web.servlet.*
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ContextConfiguration
+internal class SystemControllerTest @Autowired constructor (

```

Comment:
It's always good to have test coverage for the functionality you implement, this one is a controller test and you should also look to make tests for the SystemService class (this will be four classes or as many needed depending on the functionary you have as discussed in the previous comment). 

You don't need to inject dependencies into this class, most of the time you don't but here no need.

You can have it like.

```
@WebMvcTest(controllers = [SystemController::class])
@ActiveProfiles("test")
internal class SystemControllerTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

}
``` 

ObjectMapper is not a dependency for the class so I'm not sure why it's injected.

Reply:
I tried this but couldn't get it to compile so I decided to leave it with the original format for now.  

I think the ObjectMapper dependency is used in the Controller Test classes when I am mapping Data Transfer Objects as strings when testing post requests. 

Have a look and see what you think.  It may be that I am not using it right.